lib/src/main/kotlin/com/swmansion/starknet/data/TransactionHashCalculator.kt

```diff
@@ -0,0 +1,97 @@
+package com.swmansion.starknet.data
+
+import com.swmansion.starknet.crypto.StarknetCurve
+import com.swmansion.starknet.data.types.Calldata
+import com.swmansion.starknet.data.types.Felt
+import com.swmansion.starknet.data.types.StarknetChainId
+import com.swmansion.starknet.data.types.transactions.TransactionType
+
+/**
+ * Toolkit for calculating hashes of transactions.
+ */
+object TransactionHashCalculator {
```

Comment:
I hate the name of this and ContractAddressCalculator and I would gladly accept suggestions.

Reply:
@THenry14 any ideas?

Reply:
Nah, that's why I liked Artur's comment